source/pod.go

```diff
@@ -146,6 +150,16 @@ func (ps *podSource) Endpoints(ctx context.Context) ([]*endpoint.Endpoint, error
 				}
 			}
 		}
+
+		if ps.podSourceDomain != "" {
+			if len(targets) == 0 {
+				addToEndpointMap(endpointMap, pod.ObjectMeta.Name+"."+ps.podSourceDomain, suitableType(pod.Status.PodIP), pod.Status.PodIP)
+			} else {
+				for _, target := range targets {
+					addToEndpointMap(endpointMap, pod.ObjectMeta.Name+"."+ps.podSourceDomain, suitableType(target), target)
+				}
+			}
+		}
```

Comment:
```suggestion
		if ps.podSourceDomain != "" {
			domain := pod.ObjectMeta.Name+"."+ps.podSourceDomain
			for _, target := range targets {
				addToEndpointMap(endpointMap, domain, suitableType(target), target)
			}		 
			if len(targets) == 0 {
				addToEndpointMap(endpointMap, domain, suitableType(pod.Status.PodIP), pod.Status.PodIP)
			}
		}
```
Wdyt of re-writing it like this ? I tried to be more readable and avoid multiple if.